### ISSUE TITLE
CSharpConvertIfToSwitchCodeRefactoringProvider crashes in some Roslyn code

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#22266]
In `SymbolExtensions.CheckUnsafeModifier`, click on `if`. I get the yellow band for analyzer crash with stacktrace which I copied below.

```C#
        internal static void CheckUnsafeModifier(this Symbol symbol, DeclarationModifiers modifiers, Location errorLocation, DiagnosticBag diagnostics)
        {
            if (((modifiers & DeclarationModifiers.Unsafe) == DeclarationModifiers.Unsafe) && !symbol.CompilationAllowsUnsafe())
            {
                Debug.Assert(errorLocation != null);
                diagnostics.Add(ErrorCode.ERR_IllegalUnsafe, errorLocation);
            }
        }
```

```
System.InvalidCastException : Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.ParenthesizedExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.BinaryExpressionSyntax'.
   at Microsoft.CodeAnalysis.CSharp.ConvertIfToSwitch.CSharpConvertIfToSwitchCodeRefactoringProvider.CSharpAnalyzer.CreatePatternFromExpression(ExpressionSyntax operand)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.ConvertIfToSwitch.AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer`4.<GetSections>d__7.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at async Microsoft.CodeAnalysis.ConvertIfToSwitch.AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer`4.ComputeRefactoringsAsync[TStatementSyntax,TIfStatementSyntax,TExpressionSyntax,TSwitchLabelSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ConvertIfToSwitch.AbstractConvertIfToSwitchCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by copa017** [comment]
Here it crashes too on 
```
if ((currentSessionID == 0) && (newCurrentSessionId == 0))
                        { ....
```
(both variables are simple 'int')

**Comment by skorunka** [comment]
Have seen it in the VS 2017 IDE few times `Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.ParenthesizedExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.BinaryExpressionSyntax'.` Not sure what exact line code causing it.

**Comment by jinujoseph** [comment]
also mentioned at https://developercommunity.visualstudio.com/content/problem/142350/vs2017-1542-csharpconvertiftoswitchcoderefactoring.html

**Comment by CyrusNajmabadi** [comment]
I believe this was fixed with https://github.com/dotnet/roslyn/pull/21362

**Comment by jinujoseph** [comment]
also mentioned [Link](https://developercommunity.visualstudio.com/content/problem/142896/%E1%84%8F%E1%85%A9%E1%84%83%E1%85%B5%E1%86%BC-%E1%84%8C%E1%85%AE%E1%86%BC-%E1%84%92%E1%85%AA%E1%84%86%E1%85%A7%E1%86%AB-%E1%84%89%E1%85%A1%E1%86%BC%E1%84%83%E1%85%A1%E1%86%AB%E1%84%8B%E1%85%A6-%E1%84%82%E1%85%A9%E1%84%85%E1%85%A1%E1%86%AB%E1%84%89%E1%85%A2%E1%86%A8-%E1%84%87%E1%85%A1%E1%84%80%E1%85%A1-%E1%84%82%E1%85%A1%E1%84%90%E1%85%A1%E1%84%82%E1%85%A1%E1%84%86%E1%85%A7%E1%86%AB%E1%84%89%E1%85%A5.html)

**Comment by chborl** [comment]
Thanks for the heads up, @CyrusNajmabadi.  I confirmed that this is fixed in 15.5 by [21362](https://github.com/dotnet/roslyn/pull/21362)    

I wasn't able to reproduce the bug on 15.5 so I undid the code change from #[21362](https://github.com/dotnet/roslyn/pull/21362) and was able to repro it there.  Also, I verified that 15.4.2 did not contain this fix, but 15.5 does.
